### ISSUE TITLE
return may reach end of non-void function

### ISSUE THREAD

**Issue by defunctzombie** [ros/common_msgs#84]
When building with `-Wall Werror` on the latest version of indigo I am hit with the following error during compilation:

``` shell
/opt/ros/indigo/include/sensor_msgs/point_field_conversion.h:123:5: error: control may reach end of non-void function [-Werror,-Wreturn-type]
    }
```

**Compiler**:
clang 3.6

**OS**:
Ubuntu 14.04

This prevents my project from using `-Wall -Werror`.

I can make a patch but it is unclear to me how you would prefer the error condition be handled here.

**Comment by tfoote** [comment]
Indeed the "error" case should not be hit as long as we have full coverage of the valid datatypes. Maybe an assert followed by a return of the default value to make the compiler happy.

**Comment by defunctzombie** [comment]
@tfoote what would you consider the default value to be?

And for assert, would I use the builtin system asserts or is there a more "ROS" like way that is preferred?

**Comment by alspitz** [comment]
Hitting this on kinetic as well.

**Comment by tfoote** [comment]
In this case a ROS_BREAK would be recommended: http://wiki.ros.org/CppStyleGuide#Assertions

You can just make a T and return it with it's default value I think since we should never get there it won't matter the value.

**Comment by alspitz** [comment]
Would adding a dependency on rosconsole be needed / all right?

**Comment by jasonimercer** [comment]
Any thoughts on throwing a `std::invalid_argument` exception? This gets interpreted as a correct return path (compiler doesn't warn), has more semantic value than an abort and doesn't require adding a dummy return value.

**Comment by tfoote** [comment]
That would also be a valid approach. We couldn't add that in mid release though. And we'd have to make sure to document the possible exception at each of the levels of API above this.

**Comment by tfoote** [comment]
Warning addressed in #118
